stockflux-securities-master/src/components/InputForm.js

```diff
@@ -30,75 +59,90 @@ const InputForm = ({ match }) => {
-          />
-        </div>
-        <div className="input-row">
-          <label className="input-label" htmlFor="exchange-input">
-            Exchange
-          </label>
-          <input
-            className="input-form-input"
-            id="exchange-input"
-            value={exchange}
-            onChange={event => setExchange(event.target.value)}
-          />
-        </div>
-        <div className="input-row">
-          <label className="input-label" htmlFor="symbol-input">
-            Symbol
-          </label>
-          <input
-            className="input-form-input"
-            id="symbol-input"
-            value={symbol}
-            onChange={event => setSymbol(event.target.value)}
-          />
-        </div>
-        <div className="input-checkbox-container">
-          <label className="input-label" htmlFor="is-shown-toggle">
-            Is Shown
-          </label>
-          <input
-            className="input-form-toggle"
-            id="is-shown-toggle"
-            value={isShown}
-            type="checkbox"
-            onChange={() => setIsShown(!isShown)}
-          />
-          <label className="toggle-switch" htmlFor="is-shown-toggle" />
-        </div>
-        <div className="input-checkbox-container">
-          <label className="input-label" htmlFor="is-enabled-toggle">
-            Is Enabled
-          </label>
-          <input
-            className="input-form-toggle"
-            id="is-enabled-toggle"
-            value={isEnabled}
-            type="checkbox"
-            onChange={() => setIsEnabled(!isEnabled)}
-          />
-          <label className="toggle-switch" htmlFor="is-enabled-toggle" />
-        </div>
-        <div className="input-submit-button-container">
-          <div
-            className={(inProgress ? "in-progress " : "") + "input-submit-button"}
-          >
-            <button>Create</button>
+      <div className="input-form-title">
+        {match.params.securityId ? "Edit Security" : "Create a Security"}
+      </div>
+      {isLoading ? (
+        <Components.LargeSpinner />
+      ) : errorMessage ? (
+        <ErrorMessage errorMessage={errorMessage} />
+      ) : (
+        <form className="input-form-body" onSubmit={submitForm}>
+          <div className="input-row">
+            <label className="input-label" htmlFor="name-input">
+              Name
+            </label>
+            <input
+              className="input-form-input"
+              id="name-input"
+              value={name}
+              onChange={event => setName(event.target.value)}
+            />
           </div>
-        </div>
-      </form>
+          <div className="input-row">
+            <label className="input-label" htmlFor="exchange-input">
+              Exchange
+            </label>
+            <input
+              className="input-form-input"
+              id="exchange-input"
+              value={exchange}
+              readOnly={match.params.securityId}
```

Comment:
Similar to our discussion yesterday with the `{condition && <Component />}` thing, I'm not sure this looks right. It will probably work fine since if there's a `securityId` then the value will be *truthy* and if there isn't one it will be `undefined` or `null` and therefore *falsy*. Is this something that you have been advised to do before? Personally, I would feel more comfortable casting it to boolean by wrapping it in `Boolean(match.params.securityId)` or `!!match.params.securityId`. I'd be interested to know what the most common approach is though.

Reply:
Following our discussion with Matt, I've changed them both to `!!match.params.securityId` .